test/widgets/profile_test.dart

```diff
@@ -79,18 +91,50 @@ CustomProfileField mkCustomProfileField(
 void main() {
   TestZulipBinding.ensureInitialized();
 
-  group('ProfilePage', () {
-    testWidgets('page builds; profile page renders', (tester) async {
-      final user = eg.user(userId: 1, fullName: 'test user',
-        deliveryEmail: 'testuser@example.com');
+  testWidgets('page builds; profile page renders', (tester) async {
+    final user = eg.user(userId: 1, fullName: 'test user',
+      deliveryEmail: 'testuser@example.com');
 
-      await setupPage(tester, users: [user], pageUserId: user.userId);
+    await setupPage(tester, users: [user], pageUserId: user.userId);
 
-      check(because: 'find user avatar', find.byType(Avatar).evaluate()).length.equals(1);
-      check(because: 'find user name', find.text('test user').evaluate()).isNotEmpty();
-      check(because: 'find user delivery email', find.text('testuser@example.com').evaluate()).isNotEmpty();
-    });
+    check(because: 'find user avatar', find.byType(Avatar).evaluate()).length.equals(1);
+    check(because: 'find user name', find.text('test user').evaluate()).isNotEmpty();
+    check(because: 'find user delivery email', find.text('testuser@example.com').evaluate()).isNotEmpty();
+  });
 
+  testWidgets('page builds; error page shows up if data is missing', (tester) async {
+    await setupPage(tester, pageUserId: eg.selfUser.userId + 1989);
+    check(because: 'find no user avatar', find.byType(Avatar).evaluate()).isEmpty();
+    check(because: 'find error icon', find.byIcon(Icons.error).evaluate()).isNotEmpty();
+  });
+
+  testWidgets('page builds; dm links to correct narrow', (tester) async {
```

Comment:
nit:

    Most of the tests in this ProfilePage group are about custom profile
    fields. Pull out the two that aren't, and rename the group.

s/two/handful/ or some other update :slightly_smiling_face: (after https://github.com/zulip/zulip-flutter/pull/1631#discussion_r2190923149)